authentication/auth_method_azure_cli_token.go

```diff
@@ -22,28 +31,28 @@ type azureCliTokenAuth struct {
 func (a azureCliTokenAuth) build(b Builder) (authMethod, error) {
 	auth := azureCliTokenAuth{
 		profile: &azureCLIProfile{
-			clientId:       b.ClientID,
-			environment:    b.Environment,
 			subscriptionId: b.SubscriptionID,
 			tenantId:       b.TenantID,
+			// Setting a non-empty string as client ID to make adal.NewServicePrincipalTokenFromManualToken() happy
+			// (See: https://github.com/Azure/go-autorest/blob/master/autorest/adal/token.go#L473)
+			// The returned ServicePrincipalToken will only use that client ID when using the default logic to refresh
+			// the token, however, as we are using custom function to refresh token (i.e. using az cli) instead, it is
+			// not necessary to pass in a valid client ID.
+			// Actually there is no formal way to get either the client ID or refresh token from Az CLI.
+			clientId:       "cli",
```

Comment:
This'll cause issues in client configurations where a users trying to pass the Client ID in - as such we should probably hard-code the Client ID of the Azure CLI here instead - which clearly isn't ideal.

Reply:
I feel for cli authentication, it makes less sense to allow user to specify anything other than subscription id. I'm not sure, but is there any use case that user need to specify a subscription ID together with tenant ID/client ID for authentication using az cli?

Reply:
I understand your concern now, I updated the code to hardcoded that cli client id inline.

Reply:
I agree with tom we shouldn't change the behaviour here and use the CLI's client ID